bigquery_etl/cli/metadata.py

```diff
@@ -0,0 +1,72 @@
+"""bigquery-etl CLI metadata command."""
+from pathlib import Path
+from typing import Optional
+
+import click
+
+from bigquery_etl.metadata.parse_metadata import DatasetMetadata, Metadata
+
+from ..cli.utils import paths_matching_name_pattern, project_id_option, sql_dir_option
+
+
+@click.group(
+    help="""
+        Commands for managing bqetl metadata.
+        """
+)
+@click.pass_context
+def metadata(ctx):
+    """Create the CLI group for the metadata command."""
+    pass
+
+
+@metadata.command(
+    help="""
+    Update metadata yaml files.
+    Updates workgroup access metadata based on the dataset_metadata.yaml and
+    deprecation metadata.
+
+    Example:
+     ./bqetl metadata update ga_derived.downloads_with_attribution_v2
+    """,
+    context_settings=dict(
+        ignore_unknown_options=True,
+        allow_extra_args=True,
+    ),
+)
+@click.argument("name")
+@project_id_option()
+@sql_dir_option
+def update(name: str, sql_dir: Optional[str], project_id: Optional[str]) -> None:
+    """Update metadata yaml file."""
+    table_metadata_files = paths_matching_name_pattern(
+        name, sql_dir, project_id=project_id, files=["metadata.yaml"]
+    )
+    dataset_metadata_path = None
+    # create and populate the dataset metadata yaml file if it does not exist
+    for table_metadata_file in table_metadata_files:
+        dataset_metadata_path = (
+            Path(table_metadata_file).parent.parent / "dataset_metadata.yaml"
+        )
+        if not dataset_metadata_path.exists():
+            continue
+        dataset_metadata = DatasetMetadata.from_file(dataset_metadata_path)
+        table_metadata = Metadata.from_file(table_metadata_file)
+
+        # set dataset metadata default_table_workgroup_access to table_workgroup_access if not set
+        if not dataset_metadata.default_table_workgroup_access:
+            dataset_metadata.default_table_workgroup_access = (
+                dataset_metadata.workgroup_access
+            )
```

Comment:
We should persist updates to `dataset_metadata.yaml`
```suggestion
            )
            dataset_metadata.write(dataset_metadata_path)
```

Reply:
Alternative that rewrites the file a bunch of times:
```suggestion
            )
        dataset_metadata.write(dataset_metadata_path)
```

Reply:
I have removed this line of code to avoid redundancy.

Reply:
I don't think this line is redundant. In earlier reviews I thought it was (or at least unnecessary), but from my testing, it's required for generated `dataset_metadata.yaml`  files to have `default_table_workgroup_access`. Without this line, those values are used by bqetl logic correctly but aren't persisted to the generated branches. This line (versus the line above) causes a lot of unnecessary writes to the file, but that's realistically fine.

I want this metadata in the generated branches because it's leveraged by ops logic to determine which ACLs are safe to propagate automatically.

Reply:
Thank you for the clarification!